examples/sampling/graphbolt/link_prediction.py

```diff
@@ -78,6 +78,40 @@ def forward(self, blocks, x):
                 hidden_x = F.relu(hidden_x)
         return hidden_x
 
+    def inference(self, graph, features, dataloader, device):
+        """Conduct layer-wise inference to get all the node embeddings."""
+        feature = features.read("node", None, "feat")
+
+        buffer_device = torch.device("cpu")
+        # Enable pin_memory for faster CPU to GPU data transfer if the
+        # model is running on a GPU.
+        pin_memory = buffer_device != device
+
+        print("Start node embedding inference.")
+        for layer_idx, layer in enumerate(self.layers):
+            is_last_layer = layer_idx == len(self.layers) - 1
+
+            y = torch.empty(
+                graph.total_num_nodes,
+                self.hidden_size,
+                dtype=torch.float32,
+                device=buffer_device,
+                pin_memory=pin_memory,
+            )
+            feature = feature.to(device)
```

Comment:
Put the entire feature in GPU could lead to OOM.

Reply:
Since all computation in CUDA device, we have to put the `feat` in cuda. To avoid GPU OOM, we use CPU offloading in the process of reassembling `y`.

Reply:
If you already put all `feat` in GPU memory, then CPU offloading is meaningless.

Reply:
I think the CPU offloading aims to the temporary tensor `y` rather than the `feat`.